src/components/Avatar/index.tsx

```diff
@@ -0,0 +1,44 @@
+import classnames from 'classnames';
+import React, { FC, useState } from 'react';
+
+import styles from './Avatar.module.css';
+
+import { IAvatarProps } from './IAvatar.d';
+
+const Avatar: FC<IAvatarProps> = ({
+  alt,
+  src,
+  size = 'sm',
+  className,
+  children,
+}) => {
+  const [error, setError] = useState<boolean>(false);
+  const textClasses = classnames(
+    styles.shared,
+    styles[`size-${size}`],
+    styles['variant-text-shared'],
+    styles[`variant-text-${size}`],
+    className
+  );
+
+  const imageClasses = classnames(
+    styles.shared,
+    styles[`size-${size}`],
+    className
+  );
+
+  return error || !src ? (
```

Comment:
I guess we should also consider children if exist, it won't make sense if the user wants to render an icon and he will pass an image src at the same time

```suggestion
  return children || !src || error ? (
```